cogs/countdown.py

```diff
@@ -1,31 +1,54 @@
 import logging
 import datetime
 import asyncio
+import random
 from discord.ext import tasks, commands
-from constants import ANNOUNCEMENTS_CHAN_ID
+from constants import ANNOUNCEMENTS_CHAN_ID, HYPE_MSGS
 
 logger = logging.getLogger(__name__)
 
 
 class Countdown(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
-        year = datetime.datetime.now().year
-        self.bitcoin_pizza_day = datetime.date(year, 5, 22)  # May 22, 2021 etc
+        year = datetime.datetime.utcnow().year
+        self.bitcoin_pizza_day = datetime.date(year, 5, 22)
+        self.tau_day = datetime.date(year, 6, 28)
         self.target_hour = 15  # 3pm UTC / 11am EDT / 10am CDT / 8am PDT
-        self.message = "Hello @everyone!! Today is {date}, which means there are **{days} DAYS until May 22**, Bitcoin Pizza Day! Let's goooo!"
+        self.messages = {
+            "0522": "Hello @everyone! Today is {date}, which means there are **{days} DAYS until May 22nd**, Bitcoin Pizza Day! {hype}",
+            "0628": "Hello @everyone! Today is {date}, which means there are **{days} DAYS until June 28th**, Tau Day! {hype}",
+            "default": "Oops, looks like my circuits got mixed up. Today is {date}; days is {days}; hype message is {hype}. Haaalp debug me!",
+        }
         self.announcement.start()
 
     def cog_unload(self):
         self.announcement.cancel()
 
     @tasks.loop(hours=24)
     async def announcement(self):
+        # bitcoin pizza day, tau day, or next year
         today = datetime.date.today()
-        days = (self.bitcoin_pizza_day - today).days
+        target_day = datetime.date(1959, 10, 31)
```

Comment:
What is the purpose of this. Looks like this value is never actually used

Reply:
yeah, it was just a little easter egg. it's neal stephenson's bday. was gonna use next year as the default and thus no need for the else statement but thought that it might confuse ppl to have that be the first default value. really we don't need one but figured, eh, why not? but if it's too much, will remove

Reply:
I'm all for fun in code. If you're trying to use that then use a constant.
```python
neal_stephensons_bday = datetime.date(1959, 10, 31)
...
target_day = neal_stephensons_bday
```

Reply:
This part of the code is never really needed.

Doesn't need to be a super generic function, but this is simply unnecessary now.
Because of the <42 days check, all this magic could go.